### PR TITLE
[freeze] fips_mode config option

### PR DESCRIPTION
**Pull request by dwoz**
### What issues does this PR fix or reference?
Fixes: #59427 

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Just needs a changelog entry

**Review by Ch3LL — DISMISSED** [review]

File: tests/unit/modules/test_x509.py [hunk 1]
```diff
@@ -436,3 +434,65 @@ def test_revoke_certificate_with_crl(self):
...
+        3lwqTx6vwK1ucS18B+nt2qqyq9hL0UvpSB7gH4KeCwCMDIfRMsrPi32jg1RyKftD
+        B+O0S5LeuJw=
+        -----END CERTIFICATE-----
+        """
+        )
+        ret = x509.read_certificate(cet)
+        assert "MD5 Finger Print" in ret
+
+
+@skipIf(not bool(pytest), False)
```
Comment (s0undt3ch) [inline tests/unit/modules/test_x509.py:467]:
```suggestion
```
Pytest is always available now.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by twangboy — DISMISSED** [review]

File: tests/unit/modules/test_x509.py (hunk 1 above)
Comment (dwoz) [reply tests/unit/modules/test_x509.py:467]: I plan to convert these tests to pytest in the master branch. I will tackle this then.

Comment (s0undt3ch) [reply tests/unit/modules/test_x509.py]: Sure, but pytest is also always available on the freeze branch.

**Review by s0undt3ch — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**